kratos/utilities/volume_inside_voxel_utility.cpp

```diff
@@ -0,0 +1,393 @@
+        PointsArrayType Nodes = rFace.Points(); 
+        double FaceArea = TetraVolume(Nodes);
+        std::vector<std::pair<double,double>> MinDistanceToNode(Edges.size(),{1,1}); 
+        
+        int NodesInside = 0;
+        for (int i = 0; i < Nodes.size(); i++) {
+            if (Nodes[i].GetSolutionStepValue(DISTANCE) > 0) NodesInside++;
+        }
+
+        if(NodesInside == 3) {
+            for (int i = 0; i < Nodes.size(); i++)  {
+                Nodes[i].GetSolutionStepValue(DISTANCE) = (-1)*Nodes[i].GetSolutionStepValue(DISTANCE);
+            }
+            double Area = HexaVolume2D(rFace,rTriangles);
+            for (int i = 0; i < Nodes.size(); i++) {
+                Nodes[i].GetSolutionStepValue(DISTANCE) = (-1)*Nodes[i].GetSolutionStepValue(DISTANCE);
+            }
+            return 1-Area;
+        }
+
+        std::vector<double> Length(Edges.size()); 
+        for(int i = 0; i < Edges.size(); i++) {
+            PointsArrayType ends = Edges[i].Points();
+            double l = Distance(ends[0], ends[1]);
+            Length[i] = l;
+        }
+
+        for (int i = 0; i < rTriangles.size(); i++) {
+            int Result = 0; 
+            array_1d<double,3> Intersection;
+            int j = 0;
+            while(!Result && j < Edges.size()) { 
+                PointsArrayType ends = Edges[j].Points();
+                Result = IntersectionUtilities::ComputeTriangleLineIntersection(rTriangles[i],ends[0],ends[1],Intersection);
+
+                if (Result) {
+                    double Dist = Distance(ends[0], Intersection);
+                    if ( Dist < (MinDistanceToNode[j].first*Length[j])) {
+                        MinDistanceToNode[j].first = Dist/Length[j];
+                    } 
+
+                    double Dist2 = Distance(ends[1], Intersection);
+                    if (Dist2 < (MinDistanceToNode[j].second*Length[j])) {
+                        MinDistanceToNode[j].second = Dist2/Length[j];
+                    } 
+                }
+                j++;
+            }
+        }
+
+        std::vector<std::vector<double>> Neighbours{{3,1},{0,2},{1,3},{2,0}};  
+        for(int i = 0; i < Nodes.size(); i++ ) {
+            array_1d<double,3> v_left{Nodes[i].X() -Nodes[Neighbours[i][0]].X(), Nodes[i].Y() -Nodes[Neighbours[i][0]].Y(), Nodes[i].Z() -Nodes[Neighbours[i][0]].Z()};
+            array_1d<double,3> v_right{Nodes[i].X() -Nodes[Neighbours[i][1]].X(), Nodes[i].Y() -Nodes[Neighbours[i][1]].Y(), Nodes[i].Z() -Nodes[Neighbours[i][1]].Z()};
+
+            double Case = GetCase(Nodes, Neighbours,i);
+            double PartialArea;
+            double factor = 1;
+            double left = 0;
+            double right = 0;
+            if (Nodes[i].GetSolutionStepValue(DISTANCE) > 0) {
+                if (Case == 0) {
+                    factor = 0.5;
+                    left = MinDistanceToNode[(i+3)%4].second;
+                    right = MinDistanceToNode[i].first;
+                }
```

Comment:
```suggestion
                } else if ()....
```
This would be the syle